site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -531,7 +531,37 @@ public function changeContactInformation($queue_code) {
         $this->core->addSuccessMessage("Queue Contact Information Changed");
         return new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']));
     }
+    /**
+     * @Route("/courses/{_semester}/{_course}/office_hours_queue/send_queue_message", methods={"POST"})
+     * @AccessControl(role="LIMITED_ACCESS_GRADER")
+     * @return RedirectResponse
+     */
+    public function sendQueueMessage(): RedirectResponse {
+        if (empty($_POST['code'])) {
+            $this->core->addErrorMessage("Missing queue name");
+            return new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']));
+        }
+        $code = trim($_POST['code']);
 
+        //if we should clear the message instead of sending a new one
+        if (!empty($_POST['clear_message'])) {
+            $this->core->getQueries()->setQueueMessage($code, 'null');
+            $this->sendSocketMessage(['type' => 'update_message', 'queue_code' => $code, 'alert' => false]);
+            $this->core->addSuccessMessage("Message cleared");
+        }
+        else {
+            if (empty($_POST['socket-message'])) {
+                $this->core->addErrorMessage("Missing message");
+                return new RedirectResponse($this->core->buildCourseUrl(['office_hours_queue']));
+            }
+            $message = trim($_POST['socket-message']);
+            $this->core->getQueries()->setQueueMessage($code, $message);
+            $this->sendSocketMessage(['type' => 'update_message', 'queue_code' => $code, 'alert' => true, 'message' => $message, 'course' => $this->core->getDisplayedCourseName()]);
```

Comment:
I think we should avoid adding any content like a message in the socket message. The reason is because any user can broadcast a message to all clients and now create notifications with arbitrary content.